### DISCUSSION TITLE
Lesson 4: Technical Tools for Output Privacy. Differential privacy.

### DISCUSSION THREAD

**Discussion by IRailean** [OpenMined/courses#212] [Course: Our Privacy Opportunity] [answered]
Hello,

In this lesson Andrew introduces differential privacy concept. He's saying that if we add a random number [-100;100] to our age before submitting it to survey we will not disclose our age. 
But it's also true that someone who has the outputs of this survey can infer information about my age:
Let's say that I submitted a number in range [-100;-81], it means that my age is under 18 and this information is leaked during this survey. Is there another more sophisticated method to add randomness to my data before submitting it that will ensure that none of my private data will be available?

Thank you!

**Comment by philipjk** [comment, marked as answer]
Hi! 
I may try to answer your question. I think part of the problem is that Andrew just introduced DP, but the theory behind it is quite rigorous and its results can be proven. Differential Privacy is a "promise", made by the data curator, that you (the owner of your data) will not be affected, adversely or otherwise, by participating to the study. Differential privacy, under some assumptions, does not guarantee that no information is taken from you (if you think about it, as long as data is useful, they _have to_ release information), rather that the information being leaked is distributional (i.e. it pertains the distribution of the answers to the query) and not particular to you, or tied to you specifically.
Going back to the example, if you add a random number _r_ in the range [-100,100] to your age _a_ (let's say under 18), there is no way for the query to actually reveal you age, or that you are a minor. To verify that, let's assume your answer is -50: that number could be the result of _a_=50 and _r_=-100, or all the combinations such that _a_ + _r_ = -50. If this question was asked to a group of people instead (picture this group as the users logged in on OpenMined, for instance), what the queries would reveal is, for example, the average age of the group, since the average of the random numbers added to the real answer is 0! Note that the average age is a distributional property of the group, and is not really tied to you specifically.

Hope it helped!
Cheers

**Reply by IRailean** [reply]
Hi @philipjk. Thank you for your reply! Indeed, Andrew has just introduced DP in this lecture, in the following lectures he will explain it a little bit more. So, DP does release some information about individuals and if you request this data an infinite number of times you will be able to fully reconstruct original data.

**Reply by philipjk** [reply]
Happy to help! I would just rephrase "DP does release some information about individuals" to "DP does release some information about the distribution underlying the data of the individuals": it's a subtle difference, but it is very important and has great implications. In fact, if the result of the query would be the same (remember that in fact we are talking about distributional information) whether or not an individual is inside the sampled group, can we really say that that individual's information has been leaked? DP takes the point of view that no, we can't.

But you are making a great point regarding  the scenario of infinite (or a large number of) queries destroying privacy! In fact, this is dubbed informally as the Fundamental Theorem of Information Recovery by the same people who introduced DP! Have a look [here](https://en.wikipedia.org/wiki/Reconstruction_attack).
